src/web/mjs/connectors/DynastyScans.mjs

```diff
@@ -82,16 +86,9 @@ export default class DynastyScans extends Connector {
      *
      */
     _getPageList( manga, chapter, callback ) {
-        fetch( this.url + chapter.id, this.requestOptions )
-            .then( response => {
-                if( response.status !== 200 ) {
-                    throw new Error( `Failed to receive page list (status: ${response.status}) - ${response.statusText}` );
-                }
-                return response.text();
-            } )
-            .then( data => {
-                let pages = data.match( /var\s+pages\s*=\s*(\[.*?\])\s*;/ )[1];
-                let pageList = JSON.parse( pages ).map( p => this.url + p.image );
+        this.fetchJSON( this.url + chapter.id + '.json' )
+            .then( json => {
+                let pageList = json['pages'].map( page => this.url + page.url );
```

Comment:
May use `new URL(page.url, this.url).href`, this will cover the case that page.url already is an absolute link

Reply:
Good idea, thanks. I'll update this ASAP too.